lib/versioning/docker/index.js

```diff
@@ -1,8 +1,96 @@
-const semver = require('../semver');
+function parse(version) {
+  const [prefix, suffix] = version.split('-');
+  const release = prefix.split('.').map(Number);
+  if (release.some(Number.isNaN)) {
+    return null;
+  }
+  return { release, suffix: suffix || '' };
+}
 
-const isValid = input => semver.isVersion(input);
+function compare(version1, vervion2) {
+  const parsed1 = parse(version1);
+  const parsed2 = parse(vervion2);
+  const length = Math.max(parsed1.release.length, parsed2.release.length);
+  for (let i = 0; i < length; i += 1) {
+    const part1 = parsed1.release[i];
+    const part2 = parsed2.release[i];
+    // shorter is bigger 2.1 > 2.1.1
```

Comment:
Will there be any current case where we compare versions with different precision? And is the significance of shorter > longer used anywhere?

Reply:
This should not happen in normal case, but the logic remains in case we need it in the future.